pkg/api/server.go

```diff
@@ -262,8 +270,44 @@ func (a *ApiServer) Start() error {
 		return err
 	}
 
-	log.Infof("Starting api server on port %d", a.apiPort)
-	return a.httpServer.Serve(listener)
+	errChan := make(chan error)
+	go func() {
+		errChan <- a.httpServer.Serve(listener)
+	}()
+
+	if a.frps != nil {
```

Comment:
Consider perhaps having one small if-block like: `if a.frps == nil { return <- errChan }` instead of having a big one